src/simplify_algebra.cpp

```diff
@@ -1035,8 +1035,15 @@ struct find_split_concat
             std::find_if(args.begin(), args.end(), [&](auto i) { return i == splits.front(); });
         if(std::distance(it, args.end()) < splits.size())
             return;
+        // Don't do anything if not all operations are slice
+        // if(not std::all_of(it, it + splits.size(), [](instruction_ref x) { 
+        //     return x->get_operator().name() == "slice";
+        // }))
+            // return;
         // If the slices are not in order then stop
         if(not std::is_sorted(it, it + splits.size(), [](instruction_ref x, instruction_ref y) {
+               if(x->get_operator().name() != "slice" or y->get_operator().name() != "slice")
```

Comment:
Dont do this in the comparisons, it doesnt really make sense, and probably can break for a lot of cases.

Reply:
This code would have the effect of ignoring any non-slice operations when checking that slice ops are in order.  If you say that doesn't make sense, that's fine with me.

Reply:
> This code would have the effect of ignoring any non-slice operations when checking that slice ops are in order. 

Thats not what it does. This is a comparison function that returns true when `x` should come before `y`. To say that `x` should come before `y` if either one is not a slice doesnt make sense. Furthermore, if we have vector of `{relu, slice}` then this would be considered in order and we would continue, which is not what we want to do.